server/src/main/scala/org/http4s/server/armeria/ArmeriaServerBuilder.scala

```diff
@@ -0,0 +1,334 @@
+  def withArmeriaBuilder(customizer: ArmeriaBuilder => Unit): Self = {
+    customizer(armeriaServerBuilder)
+    this
+  }
+
+  /** Sets the idle timeout of a connection in milliseconds for keep-alive.
+    *
+    * @param idleTimeout the timeout. [[Duration.Zero]] disables the timeout.
+    */
+  def withIdleTimeout(idleTimeout: Duration): Self = {
+    armeriaServerBuilder.idleTimeoutMillis(idleTimeout.toMillis)
+    this
+  }
+
+  /** Sets the timeout of a request.
+    *
+    * @param requestTimeout the timeout. [[Duration.Zero]] disables the timeout.
+    */
+  def withRequestTimeout(requestTimeout: Duration): Self = {
+    armeriaServerBuilder.requestTimeoutMillis(requestTimeout.toMillis)
+    this
+  }
+
+  /** Adds an HTTP port that listens on all available network interfaces.
+    *
+    * @param port the HTTP port number.
+    * @see [[ArmeriaBuilder#http(java.net.InetSocketAddress)]]
+    */
+  def withHttp(port: Int): Self = {
+    armeriaServerBuilder.http(port)
+    this
+  }
+
+  /** Adds an HTTPS port that listens on all available network interfaces.
+    *
+    * @param port the HTTPS port number.
+    * @see [[ArmeriaBuilder#https(java.net.InetSocketAddress)]]
+    */
+  def withHttps(port: Int): Self = {
+    armeriaServerBuilder.https(port)
+    this
+  }
+
+  /** Sets the [[ChannelOption]] of the server socket bound by [[BackendServer]].
+    * Note that the previously added option will be overridden if the same option is set again.
+    *
+    * @see [[https://armeria.dev/docs/advanced-production-checklist Production checklist]]
+    */
+  def withChannelOption[T](option: ChannelOption[T], value: T): Self = {
+    armeriaServerBuilder.channelOption(option, value)
+    this
+  }
+
+  /** Sets the [[ChannelOption]] of sockets accepted by [[BackendServer]].
+    * Note that the previously added option will be overridden if the same option is set again.
+    *
+    * @see [[https://armeria.dev/docs/advanced-production-checklist Production checklist]]
+    */
+  def withChildChannelOption[T](option: ChannelOption[T], value: T): Self = {
+    armeriaServerBuilder.childChannelOption(option, value)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] from the specified `keyCertChainFile`,
+    * `keyFile` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(keyCertChainFile: File, keyFile: File, keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainFile, keyFile, keyPassword.orNull)
+    this
+  }
+
+  /** Configures SSL or TLS of this [[BackendServer]] with the specified `keyCertChainInputStream`,
+    * `keyInputStream` and `keyPassword`.
+    *
+    * @see [[withTlsCustomizer(scala.Function1)]]
+    */
+  def withTls(
+      keyCertChainInputStream: InputStream,
+      keyInputStream: InputStream,
+      keyPassword: Option[String]): Self = {
+    armeriaServerBuilder.tls(keyCertChainInputStream, keyInputStream, keyPassword.orNull)
```

Comment:
Does this read from these `InputStreams` immediately?  If so, maybe this should return an `F[Self]`.  If not, I'm a little concerned with the life cycle of these `InputStream`s, and maybe we should pass `Resource[F, InputStream]`.

Reply:
> Does this read from these InputStreams immediately? 

The InputStreams are consumed immediately and converted to bytes for reusing it. Armeria does not release the InputStream. `Resource[F, InputStream]` is acceptable.